modules/core/src/main/java/org/apache/ignite/internal/MarshallerContextImpl.java

```diff
@@ -368,12 +369,27 @@ public void onMappingAccepted(final MarshallerMappingItem item) {
 
     /** {@inheritDoc} */
     @Override public Class getClass(int typeId, ClassLoader ldr) throws ClassNotFoundException, IgniteCheckedException {
-        String clsName = getClassName(JAVA_ID, typeId);
+        String clsName;
+
+        ClassNotFoundException err = null;
+
+        for (byte platformId : new byte[] {JAVA_ID, DOTNET_ID}) {
```

Comment:
`JAVA_ID` should come last - in most cases, platform type will be resolved, not a Java type.

Reply:
Fixed